### PR TITLE
spades: import from homebrew/science

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/spades.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Spades < Formula
+  desc "SPAdes: de novo genome assembly"
```
Comment (ilovezfs) [inline Formula/spades.rb]: this is circumventing the audit about descriptions not starting with formula name

File: Formula/spades.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  homepage "http://bioinf.spbau.ru/spades/"
+  url "http://cab.spbu.ru/files/release3.11.0/SPAdes-3.11.0.tar.gz"
+  sha256 "308aa3e6c5fb00221a311a8d32c5e8030990356ae03002351eac10abb66bad1f"
+
+  depends_on "cmake" => :build
+  depends_on :python if MacOS.version <= :snow_leopard
+
+  needs :openmp
```
Comment (ilovezfs) [inline Formula/spades.rb]: can we remove this?

File: Formula/spades.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    inreplace "src/projects/ionhammer/config_struct.hpp", "uint", "unsigned"
+
+    mkdir "src/build" do
+      system "cmake", "..", *std_cmake_args
+      system "make", "install"
+    end
+  end
+
+  test do
+    system "#{bin}/spades.py", "--test"
```
Comment (ilovezfs) [inline Formula/spades.rb]: can we do something beyond exit code checking

File: Formula/spades.rb (hunk 1 above)
Comment (sjackman) [reply Formula/spades.rb]: Fixed.

File: Formula/spades.rb (hunk 2 above)
Comment (sjackman) [reply Formula/spades.rb]: SPAdes requires OpenMP.

> Compiling SPAdes requires GCC >= 4.7 for OpenMP 3.1 support

File: Formula/spades.rb (hunk 3 above)
Comment (sjackman) [reply Formula/spades.rb]: Yep

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Thanks @sjackman! :spades:

**Pull request closed without merging**